### PR TITLE
Spades update

### PR DESCRIPTION
**Pull request by TCLamnidis**
Update spades module to optionally accept a yml input file.

## PR checklist

Closes #XXX 

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [x] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [x] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [x] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

File: modules/nf-core/spades/meta.yml [hunk 1]
```diff
@@ -34,6 +34,13 @@ input:
       type: file
       description: |
         List of input FastQ files of size 1, originating from Oxford Nanopore technology.
+  - yml:
+      type: file
+      description: |
+        Path to yml file containing read information.
+        The raw FASTQ files listed in this YAML file must also be supplied to the illumina input channel _in addition_ to this YML.
```
Comment (jfy133) [inline modules/nf-core/spades/meta.yml:41]:
```suggestion
        The raw FASTQ files listed in this YAML file MUST be supplied to the respective illumina/pacbio/nanopore input channel(s) _in addition_ to this YML.
```

File: tests/config/test_data.config [hunk 2]
```diff
@@ -497,6 +497,9 @@ params {
             'config' {
                 ncbi_user_settings              = "${test_data_dir}/generic/config/ncbi_user_settings.mkfg"
             }
+            'yml' {
+                spades                          = "${test_data_dir}/generic/yml/spades_input_yml.yml"
+            }
```
Comment (jfy133) [inline tests/config/test_data.config:502]:
```suggestion
```

**Review by jfy133 — APPROVED** [review]
Minor changes but LGTM once tests pass with the proper YML file

**Comment by TCLamnidis** [comment]
Updated test.yml to overcome md5sum discrepancies between conda and docker/singularity.

**Pull request merged**